appcues/src/main/java/com/appcues/ui/primitive/TextPrimitive.kt

```diff
@@ -26,27 +27,35 @@ internal fun TextPrimitive.Compose(modifier: Modifier) {
     val style = LocalTextStyle.current.applyStyle(style, context, isDark)
 
     var resizedStyle by remember(style) { mutableStateOf(style) }
+    var resizedSpans by remember(style) { mutableStateOf(spans) }
     var readyToDraw by remember(style) { mutableStateOf(false) }
 
-    Text(
-        modifier = modifier
-            .clipToBounds()
-            .drawWithContent { if (readyToDraw) drawContent() },
-        text = spans.toAnnotatedString(context, isDark),
-        style = resizedStyle,
-        overflow = TextOverflow.Ellipsis,
-        onTextLayout = { textLayoutResult ->
-            // this will attempt to auto scale down single line / char text, handling cases
-            // like emoji that are sized too large to fit in side by side layouts
-            if (textLayoutResult.lineCount == 1) {
-                if (textLayoutResult.isLineEllipsized(0)) {
-                    resizedStyle = resizedStyle.copy(fontSize = resizedStyle.fontSize * TEXT_SCALE_REDUCTION_INTERVAL)
+    key(resizedStyle) {
```

Comment:
this `key` usage I'm not sure about.

The issue here is that IF the text actually does get resized down (L47-51 below), then the outer row size also needs to resize down to the new height. Since our Row has already set it's height to IntrinsicSize.Min during the initial layout, the row stays the height of the original font size and ends up causing padding to show visually above and below the actual Text content.

Somehow, we need to force the row to reset this intrinsic height whenever this downscale happens, and the `key` here will do so. However, it is also probably causing unnecessary usage for the vast majority of cases where no resizing is happening. It does work, functionally, tested locally and snapshot tests.

Reply:
here is a visual example of the issue, with the item on the left showing what happens without this key - as the row is not resized down to match the new text size. The item on the right is with this fix.

| before | after |
| --- | --- | 
| ![Screenshot 2023-09-28 at 11 23 14 AM](https://github.com/appcues/appcues-android-sdk/assets/19266448/96988ef2-cba8-4670-8baa-fd4333587638) | ![Screenshot 2023-09-28 at 11 24 14 AM](https://github.com/appcues/appcues-android-sdk/assets/19266448/568f5d38-fdd7-4d21-8abd-7cd75a1a6115) |

Reply:
it does cause the text to redraw, yes, but the problem is that the row has already set its height to IntrinsicSize.Min, using the initial intrinsic size of the text from the original font size, which was larger. As you scale own the font, that intrinsic size also scales down, but the Compose layout system does not update it - it is only set once at the initial render. It is similar to the "deleting blue items" use case described here https://medium.com/@lepicekmichal/compose-intrinsic-redraw-bug-2cdecef0f96c